### PR TITLE
Migrate existing shared tasks from apps to shared gradle plugin

### PR DESCRIPTION
**Pull request by aleffert**
@nasthagiri please review. I'm happy to walk through it with you. Groovy/gradle are super bizarre. Also, this PR might be slightly out of context since PRs will need to follow that actually integrate this with the per app build scripts.

### CODE REVIEW

**Comment by aleffert** [comment]
cc @BenjiLee

**Comment by BenjiLee** [comment]
@aleffert @nasthagiri Maybe we should have a quick walkthrough meeting (10 mins)?

**Comment by aleffert** [comment]
@hanningni fyi, this is some in progress work to make the apps share a common gradle plugin. It still needs to reviewed, but maybe we can all go through it tomorrow.

**Comment by hanningni** [comment]
@aleffert  sure.

**Comment by aleffert** [comment]
@hanningni @BenjiLee now that we've gone over this, could you please review?

File: src/main/groovy/org/edx/builder/TaskHelper.groovy [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        files.visit { file ->
+            if(!file.isDirectory()) {
+                project.exec {
+                    executable "uncrustify"
+                    args '-c', '.uncrustify', '--replace', '--no-backup', '-l', 'OC', file.file.path
+                }
+            }
+        }
+    }
+}
```
Comment (hanningni) [inline src/main/groovy/org/edx/builder/TaskHelper.groovy:66]: should we create custom Tasks for it?

**Comment by hanningni** [comment]
LGFM

File: src/main/groovy/org/edx/builder/TaskHelper.groovy (hunk 1 above)
Comment (aleffert) [inline src/main/groovy/org/edx/builder/TaskHelper.groovy:66]: A task for downloading uncrustify?

It's not a bad idea, but given that people might want to install it in different ways (I used the homebrew package manager, for example) I didn't want to bother.

**Comment by hanningni** [comment]
:+1:

**Comment by aleffert** [comment]
@BenjiLee please review

**Comment by BenjiLee** [comment]
:+1: Looks groovy.

**Pull request merged**